Training/python/split_network_2p5.py

```diff
@@ -142,11 +142,12 @@ def ConvDenseAlias(loc):
   for comp_id in range(len(net_config.comp_names)):
     comp_name = net_config.comp_names[comp_id]
     n_comp_features = net_config.n_comp_branches[comp_id]
-    input_layer_comp = Input(name="input_{}_{}".format(loc, comp_name), shape=n_comp_features)
+    input_layer_comp = Input(name="input_{}_{}".format(loc, comp_name), shape=(1,1,n_comp_features))
     input_layers.append(input_layer_comp)
+    input_layer_resh = keras.layers.Reshape(name='{}_reshape'.format(input_layer_comp.name), target_shape=(n_comp_features,))(input_layer_comp)
```

Comment:
Reshape should not be needed. A dense layer should act on the last dimension by default. But please, check it, in case something is changed in the last TF versions.

Reply:
Indeed I tested this briefly and it seems to be fine without reshaping: 

<img width="496" alt="image" src="https://user-images.githubusercontent.com/20295221/165139352-10deb53d-81c0-45fc-8b32-016749d38963.png">

 However, reshaping with the current setup won't be compatible with [DeepTauId.cc#L1833](https://github.com/cms-tau-pog/cmssw/blob/5b56ea4f85b845ed3238c59310f79185ff481ab5/RecoTauTag/RecoTau/plugins/DeepTauId.cc#L1833) (if I understood `features.tensor<float, 4>()` operation correctly), so it should be either another reshaping done in the end of the inner/outer model, or the initial one removed.

Reply:
Hi, yes, reshape is not needed. I added it because it makes weight copying easiier

Reply:
I removed the reshape layer in https://github.com/cms-tau-pog/TauMLTools/pull/103/commits/2a5c54f53ad6a9aab11cbcd7d214b6487101bbb5